config.go

```diff
@@ -383,10 +383,28 @@ func GetDNSConfig() (*tailcfg.DNSConfig, string) {
 		if viper.IsSet("dns_config.nameservers") {
 			nameserversStr := viper.GetStringSlice("dns_config.nameservers")
 
-			nameservers := make([]netip.Addr, len(nameserversStr))
-			resolvers := make([]*dnstype.Resolver, len(nameserversStr))
+			nameservers := []netip.Addr{}
+			resolvers := []*dnstype.Resolver{}
+
+			for _, nameserverStr := range nameserversStr {
+				// Search for a nextdns pattern
+				if strings.HasPrefix(nameserverStr, "nextdns:") {
+					nextdnsID := strings.TrimPrefix(nameserverStr, "nextdns:")
+
+					if len(nextdnsID) > 0 {
+						resolvers = append(resolvers, &dnstype.Resolver{
+							Addr: fmt.Sprintf("https://dns.nextdns.io/%s", nextdnsID),
+						})
+					} else {
+						log.Error().
+							Str("func", "getDNSConfig").
+							Msgf("Could not parse nextdns nameserver: should match pattern 'nextdns:<your-id>'")
+					}
+
+					continue
+				}
```

Comment:
From what I understand, this logic is to convert:

`nextdns:abcd123` => `https://dns.nextdns.io/abcd123` ?

I am not really that keen on hardcoding dedicated providers into the config handling, this sounds like it could grow out of hand if NextDNS starts getting competitors in the DoH space.

It sounds like rather documenting how one can enter `https://dns.nextdns.io/abcd123` and make sure that experience is smoother is a better approach.

Reply:
That's fair and I can agree with that.

So this if statement could just look for a `https://` prefix instead and put it into the resolvers slice as is. Then in `getMapResponseDNSConfig` anything specific can be done that is needed for adding NextDNS machine metadata (but in another function/interface like mentioned in the other comment).

Would you be ok with that?

Reply:
Yes in `getMapResponseDNSConfig`, we can call a function which modifies the resolver map.

`addNextDNSMetadata(resolvers *dnsConfig.Resolvers)` or something more appropriate.

Reply:
Got it! I'll modify this soon.